src/Microsoft.ApplicationInsights/Extensibility/W3C/W3CActivityExtensions.cs

```diff
@@ -79,34 +62,11 @@ public static Activity UpdateContextOnActivity(this Activity activity)
         /// <param name="activity">Activity to read W3C context from.</param>
         /// <returns>traceparent header value.</returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("Activity from System.Diagnostics.DiagnosticSource 4.6.0 onwards natively support W3C.")]
         public static string GetTraceparent(this Activity activity)
         {
-            string version = null, traceId = null, spanId = null, sampled = null;
-            foreach (var tag in activity.Tags)
-            {
-                switch (tag.Key)
-                {
-                    case W3CConstants.TraceIdTag:
-                        traceId = tag.Value;
-                        break;
-                    case W3CConstants.SpanIdTag:
-                        spanId = tag.Value;
-                        break;
-                    case W3CConstants.VersionTag:
-                        version = tag.Value;
-                        break;
-                    case W3CConstants.SampledTag:
-                        sampled = tag.Value;
-                        break;
-                }
-            }
-
-            if (traceId == null || spanId == null || version == null || sampled == null)
-            {
-                return null;
-            }
-
-            return string.Join("-", version, traceId, spanId, sampled);
+            // returning activity id as there is no native way to obtain exact TraceParent header from Activity.
```

Comment:
comment is not correct, activity.Id returns id formatted as  traceparent - it's a native way to format it